qiskit/visualization/state_visualization.py

```diff
@@ -97,19 +127,178 @@ def plot_state_hinton(rho, title='', figsize=None, ax_real=None, ax_imag=None):
             qc = QuantumCircuit(2, 2)
             qc.h(0)
             qc.cx(0, 1)
-            qc.measure([0, 1], [0, 1])
 
             backend = BasicAer.get_backend('statevector_simulator')
             job = execute(qc, backend).result()
             plot_state_hinton(job.get_statevector(qc), title="New Hinton Plot")
     """
+    try:
+        mat = DensityMatrix(rho).data
+    except QiskitError:
+        raise VisualizationError('Specified object is not a quantum state.')
+
+    num = int(np.log2(len(mat)))
+
+    # get the labels
+    column_names = [bin(i)[2:].zfill(num) for i in range(2**num)]
+    row_names = [bin(i)[2:].zfill(num) for i in range(2**num)]
+
+    fig = _plot_matrix_hinton(mat=mat, row_names=row_names, column_names=column_names,
+                              oper_name='$\\rho$', title=title, figsize=figsize,
+                              ax_real=ax_real, ax_imag=ax_imag, auto_scale=auto_scale)
+
+    if ax_real is None and ax_imag is None:
+        if get_backend() in ['module://ipykernel.pylab.backend_inline',
+                             'nbAgg']:
+            plt.close(fig)
+        return fig
+
+
+def plot_quantum_channel_hinton(
+        channel: Union[QuantumChannel, QuantumCircuit],
```

Comment:
I like accepting QuantumChannel directly but a lot of other plotting methods take ndarray. So I think we should support both for consistency. The ones being updated here (https://github.com/Qiskit/qiskit-terra/pull/4324/files) also preserve the ndarray functionality still. Maybe we deprecate this altogether but while they exist I think they should exist on all drawers.

Reply:
Thanks for the feedback. Actually the function is still able to receive ndarray, but we cannot guarantee the input is Chi matrix representation.

For example,
```
zx = quantum_info.Operator.from_label('ZX')
zx_array = zx.data

plot_quantum_channel_hinton(zx_array)
```
will output meaningless plot (not like a quantum state, there are multiple channel representations in Qiskit).

So I explicitly added ndarray to the type hint and also added a warning when the array is given.